esmvalcore/preprocessor/_multimodel.py

```diff
@@ -139,8 +141,20 @@ def _unify_time_coordinates(cubes):
         # Update the cubes' time coordinate (both point values and the units!)
         cube.coord('time').points = date2num(dates, t_unit, coord.dtype)
         cube.coord('time').units = t_unit
-        cube.coord('time').bounds = None
+        _guess_time_bounds(cube)
+
+
+def _guess_time_bounds(cube):
+    """Guess time bounds if possible."""
+    cube.coord('time').bounds = None
+    try:
         cube.coord('time').guess_bounds()
+    except ValueError:  # Time has length 1 --> guessing bounds not possible
```

Comment:
It is considered good practice to only use exceptions if really necessary. If you already know that this will only happen when the time coordinate has length 1, it would be cleaner to write:
```python
    if cube.coord('time').shape == (1,):
        logger.debug(
            "Encountered scalar time coordinate in multi_model_statistics: "
            "cannot determine its bounds"
        )
    else:
        cube.coord('time').guess_bounds()
```
that also avoids the comment explaining why you need the exception, because you can immediately see it from the code.